core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java

```diff
@@ -72,6 +72,7 @@ private Settings nodeSettings() {
         return Settings.builder()
                 .put("node.add_id_to_custom_path", false)
                 .put("node.enable_custom_paths", true)
+                .put("node.shared_data", createTempDir().getParent())
```

Comment:
since this test is contained maybe it could configure a shared data path and then make sure that all index paths are sub dirs?